examples/task_controller_server/main.cpp

```diff
@@ -0,0 +1,173 @@
+	{
+	}
+
+	bool on_value_command(std::shared_ptr<isobus::ControlFunction>, std::uint16_t, std::uint16_t, std::int32_t, std::uint8_t &)
+	{
+		return true;
+	}
+
+	bool store_device_descriptor_object_pool(std::shared_ptr<isobus::ControlFunction>, const std::vector<std::uint8_t> &, bool)
+	{
+		return true;
+	}
+};
+
+int main()
+{
+	std::signal(SIGINT, signal_handler);
+
+	std::shared_ptr<isobus::CANHardwarePlugin> canDriver = nullptr;
+#if defined(ISOBUS_SOCKETCAN_AVAILABLE)
+	canDriver = std::make_shared<isobus::SocketCANInterface>("can0");
+#elif defined(ISOBUS_WINDOWSPCANBASIC_AVAILABLE)
+	canDriver = std::make_shared<isobus::PCANBasicWindowsPlugin>(PCAN_USBBUS1);
+#elif defined(ISOBUS_WINDOWSINNOMAKERUSB2CAN_AVAILABLE)
+	canDriver = std::make_shared<isobus::InnoMakerUSB2CANWindowsPlugin>(0); // CAN0
+#elif defined(ISOBUS_MACCANPCAN_AVAILABLE)
+	canDriver = std::make_shared<isobus::MacCANPCANPlugin>(PCAN_USBBUS1);
+#elif defined(ISOBUS_SYS_TEC_AVAILABLE)
+	canDriver = std::make_shared<isobus::SysTecWindowsPlugin>();
+#endif
+	if (nullptr == canDriver)
+	{
+		std::cout << "Unable to find a CAN driver. Please make sure you have one of the above drivers installed with the library." << std::endl;
+		std::cout << "If you want to use a different driver, please add it to the list above." << std::endl;
+		return -1;
+	}
+
+	isobus::CANStackLogger::set_can_stack_logger_sink(&logger);
+	isobus::CANStackLogger::set_log_level(isobus::CANStackLogger::LoggingLevel::Debug); // Change this to Debug to see more information
+	isobus::CANHardwareInterface::set_number_of_can_channels(1);
+	isobus::CANHardwareInterface::assign_can_channel_frame_handler(0, canDriver);
+
+	if ((!isobus::CANHardwareInterface::start()) || (!canDriver->get_is_valid()))
+	{
+		std::cout << "Failed to start hardware interface. The CAN driver might be invalid." << std::endl;
+		return -2;
+	}
+
+	std::this_thread::sleep_for(std::chrono::milliseconds(250));
+
+	isobus::NAME TestDeviceNAME(0);
+
+	//! Make sure you change these for your device!!!!
+	TestDeviceNAME.set_arbitrary_address_capable(true);
+	TestDeviceNAME.set_industry_group(2);
+	TestDeviceNAME.set_device_class(0);
+	TestDeviceNAME.set_function_code(static_cast<std::uint8_t>(isobus::NAME::Function::TaskController));
+	TestDeviceNAME.set_identity_number(20);
+	TestDeviceNAME.set_ecu_instance(0);
+	TestDeviceNAME.set_function_instance(0); // TC #1. If you want to change the TC number, change this.
+	TestDeviceNAME.set_device_class_instance(0);
+	TestDeviceNAME.set_manufacturer_code(1407);
+
+	auto TestInternalECU = isobus::InternalControlFunction::create(TestDeviceNAME, 247, 0); // The preferred address for a TC is defined in ISO 11783
+	MyTCServer server(TestInternalECU, 4, 255, 16, 0x17); // 4 booms, 255 sections, 16 channels, some options configured using isobus::TaskControllerServer::ServerOptions, such as "Supports Documentation"
```

Comment:
Hmm, might be good to have some way to construct the 0x17 options more describtively to set a good example. not sure how though... maybe a simple ServerOptionsBuilder class/struct would suffice?

Reply:
Yeah bit fields have always felt weird/tricky to construct nicely.  A tiny helper class could work better than some arcane hex value I think

Reply:
See if you like the way I have it now. It looks like this:
```
MyTCServer server(TestInternalECU,
                  4, // Booms
                  255, // Sections
                  16, // Channels
                  isobus::TaskControllerOptions()
                    .with_documentation()
                    .with_implement_section_control()
                    .with_tc_geo_with_position_based_control());
```

Reply:
Perfect!